sys/include/net/nanocoap.h

```diff
@@ -567,6 +567,54 @@ static inline void *coap_get_token(const coap_pkt_t *pkt)
     return coap_hdr_data_ptr(pkt->hdr);
 }
 
+/**
+ * @brief   Get the token from a CoAP over UDP packet from a buffer
+ *
+ * @warning This function will happily read past the buffer on invalid input.
+ *          Avoid using this function. Use @ref coap_parse instead.
+ *
+ * @pre     This assumes that the buffer has already been validated (e.g. using
+ *          @ref coap_udp_parse).
+ */
+static inline void *coap_get_token_from_udp_buf(uint8_t *buf)
+{
+    coap_udp_hdr_t *hdr = (void *)buf;
+
+    uint8_t tkl = hdr->ver_t_tkl & 0xf;
+    size_t tkl_ext_len =  coap_get_ext_tkl_field_len_from_tkl_val(tkl);
```

Comment:
Where is `coap_get_ext_tkl_field_len_from_tkl_val()` defined?

Reply:
In the branch I cherry-picked this from :facepalm: